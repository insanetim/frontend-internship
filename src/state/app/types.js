export const LOADING_ON = 'app/LOADING_ON'
export const LOADING_OFF = 'app/LOADING_OFF'
export const SHOW_MODAL = 'app/SHOW_MODAL'
export const HIDE_MODAL = 'app/HIDE_MODAL'
export const SHOW_NOTIFICATION = 'app/SHOW_NOTIFICATION'
export const HIDE_NOTIFICATION = 'app/HIDE_NOTIFICATION'
