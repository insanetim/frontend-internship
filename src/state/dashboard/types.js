export const REQUEST_TRENDING = 'dashboard/REQUEST_TRENDING'
export const SET_TRENDING = 'dashboard/SET_TRENDING'
export const REQUEST_SEARCH = 'dashboard/REQUEST_SEARCH'
export const SET_SEARCH = 'dashboard/SET_SEARCH'
export const SET_SEARCH_QUERY = 'dashboard/SET_SEARCH_QUERY'
export const CLEAR_SEARCH = 'dashboard/CLEAR_SEARCH'
