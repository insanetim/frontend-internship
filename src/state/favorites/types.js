export const FETCH_FAVORITES = 'favorites/FETCH_FAVORITES'
export const SET_FAVORITES = 'favorites/SET_FAVORITES'
