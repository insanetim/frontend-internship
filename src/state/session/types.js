export const LOG_IN = 'session/LOG_IN'
export const LOG_IN_CANCEL = 'session/LOG_IN_CANCEL'
export const LOG_OUT = 'session/LOG_OUT'
export const SET_SESSION = 'session/SET_SESSION'
export const DELETE_SESSION = 'session/DELETE_SESSION'
export const FETCH_ACCOUNT = 'session/FETCH_ACCOUNT'
export const SET_ACCOUNT = 'session/SET_ACCOUNT'
