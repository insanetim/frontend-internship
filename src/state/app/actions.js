import * as types from './types'

export const loadingOn = () => ({
  type: types.LOADING_ON
})

export const loadingOff = () => ({
  type: types.LOADING_OFF
})

export const showNotification = payload => ({
  type: types.SHOW_NOTIFICATION,
  payload
})

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})
