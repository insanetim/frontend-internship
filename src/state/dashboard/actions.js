import * as types from './types'

export const requestTrending = payload => ({
  type: types.REQUEST_TRENDING,
  payload
})

export const setTrending = payload => ({
  type: types.SET_TRENDING,
  payload
})

export const requestSearch = payload => ({
  type: types.REQUEST_SEARCH,
  payload
})

export const setSearch = payload => ({
  type: types.SET_SEARCH,
  payload
})

export const setSearchQuery = payload => ({
  type: types.SET_SEARCH_QUERY,
  payload
})

export const clearSearch = () => ({
  type: types.CLEAR_SEARCH
})
