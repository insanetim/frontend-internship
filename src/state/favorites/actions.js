import * as types from './types'

export const fetchFavorites = (payload = 1) => ({
  type: types.FETCH_FAVORITES,
  payload
})

export const setFavorites = payload => ({
  type: types.SET_FAVORITES,
  payload
})
