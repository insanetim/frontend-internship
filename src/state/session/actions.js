import * as types from './types'

export const logIn = (payload, callback) => ({
  type: types.LOG_IN,
  payload,
  callback
})

export const logOut = callback => ({
  type: types.LOG_OUT,
  callback
})

export const setSession = payload => ({
  type: types.SET_SESSION,
  payload
})

export const deleteSession = () => ({
  type: types.DELETE_SESSION
})

export const fetchAccount = () => ({
  type: types.FETCH_ACCOUNT
})

export const setAccount = payload => ({
  type: types.SET_ACCOUNT,
  payload
})
