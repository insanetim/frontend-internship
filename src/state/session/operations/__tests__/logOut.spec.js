import Cookies from 'js-cookie'

import mockHttpClient from 'src/__mocks__/mockHttpClient'
import { showNotification } from 'src/state/app/actions'
import * as types from '../../types'
import { deleteSession } from '../../actions'
import logOut from '../logOut'

jest.mock('src/state/session/selectors', () => ({
  sessionIdSelector: jest.fn(() => 'session_id')
}))

jest.spyOn(Cookies, 'remove')

describe('logOut', () => {
  const dispatch = jest.fn()
  const callback = jest.fn()

  const action = {
    type: types.LOG_OUT
  }

  const url = '/authentication/session'

  const body = { data: { session_id: 'session_id' } }

  const response = {
    data: {
      success: true
    }
  }

  const beforeFunction = httpClient => () => {
    logOut.process(
      {
        httpClient,
        action,
        getState: jest.fn()
      },
      dispatch,
      jest.fn()
    )
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('has valid attributes', () => {
    expect(logOut).toMatchSnapshot()
  })

  describe('success', () => {
    const httpClient = mockHttpClient({ method: 'delete', response })

    beforeEach(beforeFunction(httpClient))

    it('calls right endpoint', () => {
      expect(httpClient.delete).toHaveBeenCalledTimes(1)
      expect(httpClient.delete).toHaveBeenCalledWith(url, body)
    })

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(1)
      expect(dispatch).toHaveBeenCalledWith(deleteSession())
    })
  })

  describe('callback', () => {
    const httpClient = mockHttpClient({ method: 'delete', response })

    const actionExt = {
      ...action,
      callback
    }

    it('calls callback', async () => {
      await logOut.process(
        {
          httpClient,
          action: actionExt,
          getState: jest.fn()
        },
        dispatch,
        jest.fn()
      )

      expect(callback).toHaveBeenCalledTimes(1)
    })
  })

  describe('failure', () => {
    const error = new Error('test/error')

    const httpClient = mockHttpClient({
      method: 'delete',
      reject: true,
      response: error
    })

    beforeEach(beforeFunction(httpClient))

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(1)
      expect(dispatch).toHaveBeenCalledWith(showNotification({ type: 'error', message: 'test/error' }))
    })
  })
})
