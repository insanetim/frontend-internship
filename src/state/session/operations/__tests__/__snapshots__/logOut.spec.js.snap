// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logOut has valid attributes 1`] = `
{
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "session/LOG_OUT",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
